src/test/java/io/strimzi/kafka/quotas/StaticQuotaCallbackTest.java

```diff
@@ -213,19 +228,6 @@ void shouldNotScheduleStorageCheckWhenCheckIntervalIsZero() {
         verify(scheduledExecutorService, times(0)).scheduleWithFixedDelay(any(), anyLong(), anyLong(), any(TimeUnit.class));
     }
 
-    @Test
-    void shouldNotScheduleStorageCheckWhenCheckIntervalIsNotProvided() {
-        //Given
-        ScheduledExecutorService scheduledExecutorService = mock(ScheduledExecutorService.class);
-        StaticQuotaCallback target = new StaticQuotaCallback(volumeSourceBuilder, scheduledExecutorService, Clock.systemUTC());
-
-        //When
-        target.configure(Map.of(StaticQuotaConfig.ADMIN_BOOTSTRAP_SERVER_PROP, "localhost:9092", BROKER_ID_PROPERTY, BROKER_ID));
-
-        //Then
-        verify(scheduledExecutorService, times(0)).scheduleWithFixedDelay(any(), anyLong(), anyLong(), any(TimeUnit.class));
-    }
```

Comment:
So, how does one disable it now?

Reply:
It can be disabled when the check-interval is set to 0.
But, this is already tested by the `shouldNotScheduleStorageCheckWhenCheckIntervalIsZero`, so I removed this test as it's by default enabled.

Reply:
So, we are essentially enabling it by default with this? That is a slightly different situation than adding a default interval.

Reply:
Yes, I know.
So maybe there should be config for enabling/disabling the plugin, with the default value to disabled?

Reply:
Maybe something like `client.quota.callback.static.plugin.enabled` ?

Reply:
Yeah, but that is quite substantial change. Do we really want it?

Reply:
So the plugin is enabled when the `client.quota.callback.static.class` is configured with the `io.strimzi.kafka.quotas.StaticQuotaCallback`
And the check-interval there stops the check if it's set to 0.

But at least for me when I was trying the plugin, I didn't know that the check-interval is by default set to 0, thus the plugin doesn't work - it was a bit confusing. If the plugin will run by default now and there will be information that the users can stop it when it's set to 0, I guess it will be maybe better.

Reply:
> So practically all we are discussing is the enabling of a thread to do some work and pester a broker for some file system stats.

As in my other comment. If `check-interval` is non-zero then we require one of the limits to be set and [throw an exception otherwise](https://github.com/strimzi/kafka-quotas-plugin/blob/d6ff6fbe97a64d438607a3cdcdeda0f49a0e1086/src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java#L182).

Reply:
Have put up a [PR](https://github.com/strimzi/kafka-quotas-plugin/pull/48) to soften that requirement, so we disable the storage check if no storage limits are set.

Reply:
Yeah, I guess that makes more sense ... have a default for the interval and use the storage check byteLimit / ratio to enabkle or disable the check. Thanks.

Reply:
Thanks :)